tasks/fullStack/app/models/shop.rb

```diff
@@ -1,4 +1,8 @@
 class Shop < ApplicationRecord
   has_many :products
   has_many :reviews, through: :products
+
+  def has_last_three_months_of_reviews?
+    reviews.months_ago(1).present? && reviews.months_ago(2).present? && reviews.months_ago(3).present?
```

Comment:
Absolutely you could! I was hoping to do this optimisation, but I ran out of time.

I'm sure the combined query would be more performant, simply because you only have to establish a DB connection once instead of thrice, and because you only have to do your object mapping once.

That being said, I'm not sure you'd notice much of a difference unless you're dealing with lots of data. You're still ultimately fetching the same number of rows.

It would be an interesting one to investigate/benchmark!